packages/shared/renderer/sagas.ts

```diff
@@ -295,11 +296,19 @@ function* handleSubmitProfileToRenderer(action: SendProfileToRenderer): any {
       getUnityInstance().LoadProfile(forRenderer)
     }
   } else {
+    const lastSentStatuses: Map<string, NewProfileForRenderer> = yield select(getLastSentProfileStatuses)
+    const lastSentProfileStatus = lastSentStatuses[userId]
+
     const forRenderer = profileToRendererFormat(profile.data, {
       baseUrl: fetchContentServerWithPrefix
     })
-    getUnityInstance().AddUserProfileToCatalog(forRenderer)
-    yield put(addedProfileToCatalog(userId, profile.data))
+
+    // Add deep equal check before submitting profile to renderer
+    if (!lastSentProfileStatus || !deepEqual(lastSentProfileStatus, forRenderer)) {
```

Comment:
I would avoid the `deepEqual` here in favor of checking that the `version` of the new profile is great than the one already in store.